lib/api/controllers/securityController.js

```diff
@@ -388,7 +390,19 @@ module.exports = function SecurityController (kuzzle) {
 
         return kuzzle.repositories.user.load(newRequestObject.data._id)
           .then(user => {
-            return kuzzle.repositories.user.persist(_.extend(user, newRequestObject.data.body), { database: { method: 'update' } });
```

Comment:
Then, if so, the problem lies within the repository code.
The UserRepository::load method [_should_ resolve to a hydrated object](https://github.com/kuzzleio/kuzzle/blob/develop/lib%2Fapi%2Fcore%2Fmodels%2Frepositories%2Frepository.js#L55).

I would rather fix the repository than putting the logic in the controller.